java/core/src/main/java/com/microsoft/bond/Deserializer.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond;
+
+import com.microsoft.bond.helpers.ArgumentHelper;
+import com.microsoft.bond.protocol.ProtocolWriter;
+import com.microsoft.bond.protocol.TaggedProtocolReader;
+
+import java.io.IOException;
+
+/**
+ * Resposible for deserialization of objects of a particular struct.
+ * @param <TStruct> the class of the deserialized struct value
+ */
+public final class Deserializer<TStruct extends BondSerializable> {
+
+    private final StructBondType<TStruct> structType;
+
+    /**
+     * Initializes a new deserializer.
+     *
+     * @param structType the type descriptor of the deserialized struct
+     */
+    public Deserializer(StructBondType<TStruct> structType) {
```

Comment:
Yes. How else would you do it, given that you need to know what class the data is being deserialized into.

---
In reply to: [123371871](https://github.com/Microsoft/bond/pull/502#discussion_r123371871) [](ancestors = 123371871)